core/src/main/java/org/apache/carbondata/core/scan/collector/impl/RestructureBasedVectorResultCollector.java

```diff
@@ -177,22 +177,25 @@ private void fillNoDictionaryData(CarbonColumnVector vector, ColumnVectorInfo co
   private void fillDataForNonExistingMeasures() {
     for (int i = 0; i < tableBlockExecutionInfos.getActualQueryMeasures().length; i++) {
       if (!measureInfo.getMeasureExists()[i]) {
+        int queryOrder = tableBlockExecutionInfos.getActualQueryMeasures()[i].getQueryOrder();
         CarbonMeasure measure = tableBlockExecutionInfos.getActualQueryMeasures()[i].getMeasure();
-        ColumnVectorInfo columnVectorInfo = allColumnInfo[i];
-        CarbonColumnVector vector = allColumnInfo[i].vector;
+        ColumnVectorInfo columnVectorInfo = allColumnInfo[queryOrder];
+        CarbonColumnVector vector = columnVectorInfo.vector;
         Object defaultValue = RestructureUtil
             .getMeasureDefaultValue(measure.getColumnSchema(), measure.getDefaultValue());
         if (null == defaultValue) {
           vector.putNulls(columnVectorInfo.vectorOffset, columnVectorInfo.size);
         } else {
           switch (measureInfo.getMeasureDataTypes()[i]) {
             case SHORT:
+              long shortValueAsLong = (long) defaultValue;
```

Comment:
1. Parsing string to Object should be done in constructor, so that it is done once.
2. Filling should not do any type conversion, but should fill directly into vector.